db/sqlexplain.c

```diff
@@ -189,14 +190,18 @@ static int print_cursor_description(strbuf *out, struct cursor_info *cinfo)
             strbuf_appendf(out, "index ");
         }
         strbuf_appendf(out, "temp_%d", cinfo->tbl);
+    } else if (cinfo->tbl < 0) {
+        strbuf_append(out, "table \"???\"");
     } else {
+        if (!explain)
+            rdlock_schema_lk();
+
         struct dbtable *db;
 
-        assert(cinfo->tbl >= 0);
         db = thedb->dbs[cinfo->tbl];
 
         if (cinfo->ix != -1) {
-            if (cinfo->ix <= db->nix) {
+            if (cinfo->ix < db->nix) {
```

Comment:
This conditional has caused me much pain

Reply:
Also see https://github.com/bloomberg/comdb2/blob/main/tests/recover_deadlock.test/runit#L107

Afterwards this code is called, and has cinfo->ix = 1 (bc we're looking at index "idx" https://github.com/bloomberg/comdb2/blob/main/tests/recover_deadlock.test/runit#L48), and now has db->nix = 1 (probably used to have db->nix = 2 before index was dropped). Then this code will go to else clause to call find_tag_schema, and return NULL from there which makes sense. But should this code be going to find_tag_schema at all? This was not a tag schema.

Reply:
Maybe lock should be encompassing describe_cursor(v, pc, &c) and print_cursor_description(query_plan_buf, &c, 0)?